src/ws/frame.rs

```diff
@@ -372,6 +374,10 @@ impl fmt::Display for Frame {
     }
 }
 
+/// A `Binary` representing a `WebSocket` message with framing.
+#[derive(Debug)]
+pub struct FramedBinary(pub Binary);
```

Comment:
`pub` allows user to manually create the instance, I think we should prohibit it?

User should only correctly framed messages.

Reply:
If I make it private or `pub(crate)` it complains about "can't leak private type" and won't compile

Reply:
Oh what I mean is that its member `pub Binary` should be just `Binary` so that user couldn't instantiate it himself like `FramedBianry(Bytes::new().into())`

Reply:
actually since the `frame` module is not `pub` and the struct is not exported it already is private and only accessible inside of `actix-web` so there is no change needed here

Reply:
In that case we should re-export it as public.
This `FramedBinary` will be leaked through `Frame::message` while it does prevents user fron instantiating it, it also prevents user from reading any documentation on it (you'll have blanked type `FramedBinary` without any information about it)
I think it would not be very nice for reader?

Reply:
done